pkg/ccl/changefeedccl/parquet.go

```diff
@@ -37,27 +38,46 @@ func newParquetWriterFromRow(
 		idx += 1
 		return nil
 	}); err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 
 	columnNames[idx] = parquetCrdbEventTypeColName
 	columnTypes[idx] = types.String
 
 	schemaDef, err := parquet.NewSchema(columnNames, columnTypes)
+	if err != nil {
+		return nil, 0, err
+	}
+	return schemaDef, numCols, nil
+}
+
+// newParquetWriterFromRow constructs a new parquet writer which outputs to
+// the given sink. This function interprets the schema from the supplied row.
+func newParquetWriterFromRow(
+	row cdcevent.Row, sink io.Writer, opts ...parquet.Option,
+) (*parquetWriter, error) {
+	schemaDef, numCols, err := newParquetSchemaDefintion(row)
 	if err != nil {
 		return nil, err
 	}
 
 	writerConstructor := parquet.NewWriter
+
 	if includeParquetTestMetadata {
+		if opts, err = addCDCTestMetadata(row, opts); err != nil {
```

Comment:
I wonder if we can get rid of includeParquetTestMetadata altogether? LIke at the call site (testfeed?) if we decide to use parquet, then.. just explicitly specify additional metadata?